Flora/Flora/View Controllers/SearchViewController.m

```diff
@@ -167,6 +174,41 @@ - (NSArray *)shuffleArray:(NSArray *)array {
     return mutableArray;
 }
 
+# pragma mark - GMSAutocompleteViewControllerDelegate
+
+// Handle the user's selection.
+- (void)viewController:(GMSAutocompleteViewController *)viewController
+didAutocompleteWithPlace:(GMSPlace *)place {
+  [self dismissViewControllerAnimated:YES completion:nil];
+  // Do something with the selected place.
+    self.locationLabel.text = place.formattedAddress;
+    NSLog(@"%f, %f", place.coordinate.latitude, place.coordinate.longitude);
+    self.lat = [[NSNumber numberWithDouble:place.coordinate.latitude] stringValue];
+    self.lon = [[NSNumber numberWithDouble:place.coordinate.longitude] stringValue];
+}
+
+- (void)viewController:(GMSAutocompleteViewController *)viewController
+didFailAutocompleteWithError:(NSError *)error {
+  [self dismissViewControllerAnimated:YES completion:nil];
+  // TODO: handle the error.
+  NSLog(@"Error: %@", [error description]);
```

Comment:
pop an error the user.

You may just want to write a global function that takes an NSError and does the popup thing, so you can just re-use it everywhere.